client/landing/stepper/declarative-flow/site-migration-flow.ts

```diff
@@ -26,6 +27,36 @@ import type { AssertConditionResult, Flow, ProvidedDependencies } from './intern
 
 const FLOW_NAME = 'site-migration';
 
+/**
+ * Redirects users coming from landing pages and "/sites" to the new migration flow.
+ */
+const useMigrationFlowRevampRedirect = () => {
+	const [ query ] = useSearchParams();
+	const fullQueryString = query.toString();
+	const ref = query.get( 'ref' ) ?? '';
+	const from = query.get( 'from' ) ?? '';
+
+	// Redirect user to the new migration flow.
+	useEffect( () => {
+		if (
+			! config.isEnabled( 'migration-flow/revamp' ) &&
+			! window.localStorage.getItem( 'migration-flow/revamp' )
+		) {
+			return;
+		}
+
+		// If user is coming from 'move-lp' and has a source site, we assume it's a WordPress site because it's already validated there.
+		if ( 'move-lp' === ref && '' !== from ) {
```

Comment:
Also moved the tests to the other flow since we're checking the flow now.

https://github.com/Automattic/wp-calypso/pull/93417/commits/deead7526d43b5a85c036ccb2c5cd22b6397f650